core/storage/classifier/gae_models.py

```diff
@@ -111,3 +116,87 @@ def create(
 
         classifier_model_instance.put()
         return instance_id
+
+
+class TrainClassifierJobModel(base_models.BaseModel):
+    """Model for storing classifier training jobs.
+
+    The id of instances of this class has the form
+    {{exp_id}}.{{random_hash_of_16_chars}}
+    """
+
+    # The ID of the algorithm used to create the model.
+    algorithm_id = ndb.StringProperty(required=True, choices=ALGORITHM_CHOICES)
```

Comment:
I'm not sure what the number of unique values has to do with it -- I think you'll need to defined "indexed=True" for an equality query to happen at all.

The use case I'm actually thinking about here is trying to inspect the datastore in prod GAE when something goes wrong. E.g. you might want to see why all the jobs with a particular algorithm_id are failing.